packages/moonbit-async.mbt.md

```diff
@@ -0,0 +1,1267 @@
+    defer file.close()
+    file.write("Hello, World!\nSecond line content")
+  }
+
+  // Read file
+  {
+    let file = @fs.open("test.txt", mode=ReadOnly)
+    defer file.close()
+    // encoding is included in standard library so you can use it directly
+    // file.read_all().binary() is Bytes
+    // file.read_all().text() is String
+    // file.read_all().json() is Json
+    let content = @encoding/utf8.decode(file.read_all().binary())
+    println("File content: \{content}")
+    println("File size: \{file.size()} bytes")
+  }
+
+  // Delete file
+  @fs.remove("test.txt")
+}
+```
+
+### Advanced File Operations
+
+```moonbit
+///|
+async fn advanced_file_ops() -> Unit {
+  let file = @fs.open("data.txt", mode=ReadWrite, create=0o644, truncate=true)
+  defer file.close()
+
+  // Write data
+  file.write_string("Line 1\nLine 2\nLine 3\n",encoding=@io.UTF8)
+
+  // Move file pointer
+  let _ = file.seek(0, mode=@fs.FromStart)
+
+  // Chunked reading
+  let buf = FixedArray::make(1024, b'0')
+  while file.read(buf) is n && n > 0 {
+    let arr = buf.unsafe_reinterpret_as_bytes()[0:n]
+    let _ = arr[:n]
+    println("Read \{n} bytes")
+  }
+}
+```
+
+### Directory Operations
+
+```moonbit
+///|
+async fn directory_operations() -> Unit {
+  // Method 1: Using File::as_dir()
+  let dir_file = @fs.open(".", mode=ReadOnly)
+  let dir = dir_file.as_dir()
+  defer dir.close()
+
+  let files = dir.read_all()
+  files.sort()
+
+  for file in files {
+    println("File: \{file}")
+  }
+}
+
+///|
+async fn directory_operations_advanced() -> Unit {
+  // Method 2: Direct directory operations
+  let files = @fs.readdir(".", sort=true, include_hidden=false)
+  for file in files {
+    println("File: \{file}")
+  }
+
+  // Create directory
+  @fs.mkdir("new_dir", permission=0o755)
+
+  // Remove directory (recursive)
+  @fs.rmdir("old_dir", recursive=true)
+
+  // Check file properties
+  if @fs.exists("config.txt") {
+    println("Config file exists")
+  }
+
+  if @fs.can_read("data.txt") {
+    println("Can read data file")
+  }
+
+  // Get absolute path
+  let abs_path = @fs.realpath("./relative/path")
+  println("Absolute path: \{abs_path}")
+}
+```
+
+### File System Utilities
+
+```moonbit
+async fn file_system_utilities() -> Unit {
+  // suppose example.txt and input.txt exists in this demo
+  let path = "example.txt"
+
+  // Check file existence and permissions
+  if @fs.exists(path) {
+    println("File exists")
+
+    if @fs.can_read(path) {
+      println("Can read file")
+    }
+
+    if @fs.can_write(path) {
+      println("Can write to file")
+    }
+
+    if @fs.can_execute(path) {
+      println("Can execute file")
+    }
+  }
+
+  // Get file type
+  let file_kind = @fs.kind(path)
+  match file_kind {
+    Regular => println("Regular file")
+    Directory => println("Directory")
+    SymLink => println("Symbolic link")
+    _ => println("Other file type")
+  }
+
+  // Convenient file operations
```

Comment:
I think this part deserves some emphasis. Is ordering of content relevant for AI?